pmlc/dialect/pxa/transforms/vectorize.cc

```diff
@@ -158,11 +157,25 @@ class Impl {
           builder.create<vector::BroadcastOp>(op.getLoc(), vecType, val);
       val = bcast.getResult();
     }
-    auto vecOp = builder.create<PxaVectorReduceOp>(
-        op.getLoc(), ArrayRef<Type>{op.getMemRefType()}, op.agg(), val,
-        op.memref(), op.map(), op.idxs());
-    op.replaceAllUsesWith(vecOp.getResult());
-    op.erase();
+    // Add vector_reduction only if the stride is 0
+    if (zeroStrideReductions.find(op.getOperation()) !=
+        zeroStrideReductions.end()) {
+      vector::ReductionOp reductionOp = builder.create<vector::ReductionOp>(
+          op.getLoc(), op.getMemRefType().getElementType(),
+          builder.getStringAttr("add"), val, ValueRange{});
```

Comment:
The 'add' here needs to match the agg of the reduce (i.e. if we are doing an add reduce, it's an add, if we are doing a mul reduce, we need a mul).  Not all of our reductions are supported unfortunately, so we probably also need to preflight to make sure there is a translation.  The reductions supported only seems to appear in the vector dialect verifier:

https://github.com/llvm/llvm-project/blob/master/mlir/lib/Dialect/Vector/VectorOps.cpp#L134

Reply:
If the parameter types and the type of the reduction are not supported by the vector::ReductionOp, the vectorization doesn't happen.